evm/src/cpu/kernel/asm/util/basic_macros.asm

```diff
@@ -229,3 +229,70 @@
     %select_bool
     // stack: max
 %endmacro
+
+%macro u32
+    %and_const(0xffffffff)
+%endmacro
+
+%macro not_u32
+    // stack: x
+    PUSH 0xffffffff
+    // stack: 0xffffffff, x
+    SUB
+    // stack: 0xffffffff - x
+%endmacro
+
+%macro add3_u32
+    // stack: x, y, z
+    ADD
+    // stack: x+y, z
+    ADD
+    // stack: x+y+z
+    %u32
+%endmacro
+
+
+// given u32 bytestring abcd return dcba
+%macro reverse_bytes_u32
```

Comment:
Did you see the `BYTE` instruction? I think it could lead to a shorter implementation